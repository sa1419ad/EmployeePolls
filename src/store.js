import { configureStore } from "@reduxjs/toolkit";
import authedUser from "./Reducers/AuthedUser";
import users from "./Reducers/Users";
import questions from "./Reducers/Questions";

export const store = configureStore({
  reducer: {
    authedUser,
    users,
    questions,
  },
});
