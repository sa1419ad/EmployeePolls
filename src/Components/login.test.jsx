import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter } from "react-router-dom";
import Login from "./login";

describe("Login", () => {
  it("should render the component", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should display all elements", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    const UsernameLable = screen.getByTestId("UserNameLable");
    const passwordLable = screen.getByTestId("passwordLable");
    expect(UsernameLable.textContent).toBe("User name");
    expect(passwordLable.textContent).toBe("Password");
  });
  it("should clear input elements after clicking submit button", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    const Usernameinput = screen.getByTestId("Username");
    const password = screen.getByTestId("password");
    const LoginKlick = screen.getByTestId("LoginKlick");

    expect(Usernameinput.value).toBe("");
    expect(password.value).toBe("");

    fireEvent.change(Usernameinput, { target: { value: "sarahedo" } });
    fireEvent.change(password, {
      target: { value: "password123" },
    });

    expect(Usernameinput.value).toBe("sarahedo");
    expect(password.value).toBe("password123");

    fireEvent.click(LoginKlick);

    expect(Usernameinput.value).toBe("sarahedo");
    expect(password.value).toBe("password123");
  });
});
