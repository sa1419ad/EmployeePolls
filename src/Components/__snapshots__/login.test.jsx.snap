// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the component 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center align-items-center"
    style="height: 100vh;"
  >
    <form
      class="mx-auto my-5 p-4 border rounded"
      style="width: 30%;"
    >
      <h1
        class="text-center mb-4"
      >
        Login
      </h1>
      <div
        class="mb-3 col"
      >
        <label
          class="form-label"
          data-testid="UserNameLable"
          for="username"
        >
          User name
        </label>
        <input
          class="form-control"
          data-testid="Username"
          id="username"
          name="username"
          placeholder="Enter User name"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="mb-3 col"
      >
        <label
          class="form-label"
          data-testid="passwordLable"
          for="password"
        >
          Password
        </label>
        <input
          class="form-control"
          data-testid="password"
          id="password"
          name="password"
          placeholder="Enter Password"
          type="password"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <button
        class="w-100 btn btn-primary"
        data-testid="LoginKlick"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
